### ISSUE TITLE
istio-citadel always CrashLoopBackOff 

### ISSUE THREAD

**Issue by niejian** [istio/istio#15363]
* version： istio: 1.2.0, k8s: 1.5.0
when I installed istio, then use ` kubectl get pods -n istio-system` to see some pods.
```bash
NAME                                      READY   STATUS              RESTARTS   AGE
grafana-6575997f54-plzrp                  1/1     Running             0          18h
istio-citadel-755f5c8f94-9nbrj            0/1     CrashLoopBackOff    215        18h
istio-cleanup-secrets-1.2.0-mnpgp         0/1     Completed           0          18h
istio-egressgateway-845598d5c7-249t5      0/1     Running             0          18h
istio-galley-6c466bf5b6-bv24v             0/1     ContainerCreating   0          18h
istio-ingressgateway-d5bd74669-7gl6n      0/1     Running             0          18h
istio-pilot-5bcff9bb4d-mz8n7              1/2     Running             131        18h
istio-pilot-7d67565bd-pbhnm               0/2     Error               133        18h
istio-policy-6c8f778c66-64grg             1/2     CrashLoopBackOff    386        18h
istio-sidecar-injector-679f5fb7c9-rj7lh   0/1     ContainerCreating   0          18h
istio-telemetry-5b79949cbc-94knx          1/2     CrashLoopBackOff    387        18h
istio-tracing-555cf644d-x46mf             1/1     Running             0          18h
kiali-6cd6f9dfb5-ln7hv                    1/1     Running             0          18h
prometheus-7d7b9f7844-nh9gc               0/1     ContainerCreating   0          18h
```
here is the log
```bash
 kubectl logs istio-citadel-755f5c8f94-9nbrj  -n istio-system
2019-07-05T01:39:30.847239Z     info    The custom-defined DNS name list is [istio-pilot-service-account.istio-system:istio-pilot.istio-system]
2019-07-05T01:39:30.847240Z     info    ControlZ available at 127.0.0.1:9876
2019-07-05T01:39:30.849087Z     info    Use self-signed certificate as the CA certificate
2019-07-05T01:39:30.849582Z     info    Failed to get secret (error: Get https://10.96.0.1:443/api/v1/namespaces/istio-system/secrets/istio-ca-secret: dial tcp 10.96.0.1:443: connect: connection refused), will create one
2019-07-05T01:39:31.445524Z     error   Failed to write secret to CA (error: Post https://10.96.0.1:443/api/v1/namespaces/istio-system/secrets: dial tcp 10.96.0.1:443: connect: connection refused). Abort.
2019-07-05T01:39:31.445559Z     error   Failed to create a self-signed Citadel (error: failed to create CA due to secret write error)
```
**I remove all components and install it again, it occur again, does any one know how to solve it**

**Comment by howardjohn** [comment]
looks like failed to connect to api server, is it crashing or something? Do you have some restrictive network policy blocking it?

**Comment by niejian** [comment]
> looks like failed to connect to api server, is it crashing or something? Do you have some restrictive network policy blocking it?

no. But it work in version 1.1.7, failed in version 1.2.0

**Comment by niejian** [comment]
it works well in version 1.2.2, any body konw how?

**Comment by morgulnet** [comment]
istio-1.4.0

NAME                                      READY   STATUS              RESTARTS   AGE
grafana-6b65874977-24fzw                  1/1     Running             0          32m
istio-citadel-f74b87856-4pzmd             0/1     CrashLoopBackOff    11         32m
istio-egressgateway-c56696f94-89n65       0/1     Running             0          32m
istio-galley-566bcc7758-d2xpq             0/1     ContainerCreating   0          32m
istio-ingressgateway-6b9987bf94-fldj8     0/1     Running             0          32m
istio-pilot-677c9c688f-wnc49              0/1     Running             0          32m
istio-policy-58fb4d586c-khb9s             0/1     CrashLoopBackOff    15         32m
istio-sidecar-injector-85657c7bcb-wcr4w   0/1     ContainerCreating   0          32m
istio-telemetry-666b985499-sw7fw          0/1     CrashLoopBackOff    15         32m
istio-tracing-c66d67cd9-85kvk             1/1     Running             0          32m
kiali-8559969566-dhtmk                    1/1     Running             0          32m
prometheus-66c5887c86-nn8kc               0/1     ContainerCreating   0          32m

2019-11-25T09:10:26.733172Z	info	The custom-defined DNS name list is [istio-galley-service-account.istio-config:istio-galley.istio-config.svc istio-galley-service-account.istio-control:istio-galley.istio-control.svc istio-galley-service-account.istio-control-master:istio-galley.istio-control-master.svc istio-galley-service-account.istio-master:istio-galley.istio-master.svc istio-galley-service-account.istio-pilot11:istio-galley.istio-pilot11.svc istio-pilot-service-account.istio-control:istio-pilot.istio-control istio-pilot-service-account.istio-pilot11:istio-pilot.istio-system istio-sidecar-injector-service-account.istio-control:istio-sidecar-injector.istio-control.svc istio-sidecar-injector-service-account.istio-control-master:istio-sidecar-injector.istio-control-master.svc istio-sidecar-injector-service-account.istio-master:istio-sidecar-injector.istio-master.svc istio-sidecar-injector-service-account.istio-pilot11:istio-sidecar-injector.istio-pilot11.svc istio-sidecar-injector-service-account.istio-remote:istio-sidecar-injector.istio-remote.svc ]
2019-11-25T09:10:26.733244Z	warn	k8sController	Cannot process this invalid custom defined names , it should follow SERVICE_ACCOUNT.NAMESPACE:DOMAIN format
2019-11-25T09:10:26.733325Z	info	ControlZ available at 127.0.0.1:9876
2019-11-25T09:10:26.737766Z	info	Use self-signed certificate as the CA certificate
2019-11-25T09:10:27.740233Z	info	pkiCaLog	Failed to get secret (error: Get https://10.233.0.1:443/api/v1/namespaces/istio-system/secrets/istio-ca-secret: dial tcp 10.233.0.1:443: connect: no route to host), will create one
2019-11-25T09:10:28.936160Z	error	pkiCaLog	Failed to write secret to CA (error: Post https://10.233.0.1:443/api/v1/namespaces/istio-system/secrets: dial tcp 10.233.0.1:443: connect: no route to host). Abort.
2019-11-25T09:10:28.936208Z	error	Failed to create a self-signed Citadel (error: failed to create CA due to secret write error)
